src/model/utils/XrefUtils.tsx

```diff
@@ -183,6 +179,32 @@ export const getUnconditionalImplicitXrefs = () => {
   return foundXrefs;
 };
 
+export const getJoinedXrefs = (xrefs: Xref[]) => {
+  const joinedXrefs = [...xrefs];
+  const scheduledForRemoval: string[] = [];
+  joinedXrefs
+    .filter(xref => xref.properties && xref.properties.Status === 'JOINED')
+    .forEach(xref => {
```

Comment:
The combination of `forEach` and `find` will make this O(n^2). I am wondering if it is possible to make this O(n) ie a single pass through the xrefs?